daml-script/test/BUILD.bazel

```diff
@@ -162,10 +163,79 @@ da_scala_library(
     ],
 )
 
+daml_compile(
+    name = "coin-v1",
+    srcs = [":upgrades/coin-v1/Coin.daml"],
+    project_name = "coin",
+    target = "1.dev",
+    version = "1.0.0",
+)
+
+daml_compile(
+    name = "coin-v2",
+    srcs = [":upgrades/coin-v2/Coin.daml"],
+    project_name = "coin",
+    target = "1.dev",
+    upgrades = "//daml-script/test:coin-v1.dar",
+    version = "2.0.0",
+)
+
+daml_compile(
+    name = "coin-v3",
+    srcs = [":upgrades/coin-v3/Coin.daml"],
+    project_name = "coin",
+    target = "1.dev",
+    upgrades = "//daml-script/test:coin-v2.dar",
+    version = "3.0.0",
+)
+
+daml_compile(
+    name = "coin-upgrade-v1-v2",
+    srcs = [":upgrades/coin-upgrade-v1-v2/CoinUpgrade.daml"],
+    data_dependencies = [
+        "//daml-script/test:coin-v1.dar",
+        "//daml-script/test:coin-v2.dar",
+    ],
+    dependencies = [
+        "//daml-script/daml:daml-script-1.dev.dar",
+    ],
+    module_prefixes = {
+        "coin-1.0.0": "Coin_1_0_0",
+        "coin-2.0.0": "Coin_2_0_0",
+    },
+    project_name = "coin-upgrade",
+    target = "1.dev",
+    version = "2.0.0",
+)
+
+daml_compile(
+    name = "coin-upgrade-v1-v3",
```

Comment:
Checking the transitive case is great! This wouldn't go into this PR, but this has gotten me wondering if we could design some tooling to expose transitive and intermediate dependencies for upgrades and visualize the chain of changes/extensions to a template/datatype.